include/deal.II/numerics/vector_tools_project.templates.h

```diff
@@ -826,6 +834,40 @@ namespace VectorTools
                      project_to_boundary_first);
         }
     }
+
+    /**
+     * Specialization of project() for complex numbers in case where we cannot
+     * use the MatrixFree implementation and instead need to use the matrix
+     * based one.
+     */
+    template <typename VectorType, int dim, int spacedim>
+        typename std::enable_if <
+        numbers::NumberTraits<typename VectorType::value_type>::is_complex ||
+      dim<spacedim, void>::type
```

Comment:
This is rather awkwardly indented. Can you see what you can achieve with parentheses?